File: Examples/Python/underautomation/universal_robots/ssh/tools/ssh_command.py
import typing
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "..", "..", "..", 'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.Ssh.Tools import SshCommand as ssh_command

class SshCommand:
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = ssh_command()
		else:
			self._instance = _internal
	def begin_execute(self) -> typing.Any:
		return self._instance.BeginExecute()
	def begin_execute(self, callback: typing.Any) -> typing.Any:
		return self._instance.BeginExecute(callback)
	def begin_execute(self, callback: typing.Any, state: typing.Any) -> typing.Any:
		return self._instance.BeginExecute(callback, state)
	def begin_execute(self, commandText: str, callback: typing.Any, state: typing.Any) -> typing.Any:
		return self._instance.BeginExecute(commandText, callback, state)
	def end_execute(self, asyncResult: typing.Any) -> str:
		return self._instance.EndExecute(asyncResult)
	def execute(self) -> str:
		return self._instance.Execute()
	def cancel_async(self) -> None:
		self._instance.CancelAsync()
	def execute(self, commandText: str) -> str:
		return self._instance.Execute(commandText)
	def dispose(self) -> None:
		self._instance.Dispose()
	@property
	def command_text(self) -> str:
		return self._instance.CommandText
	@property
	def command_timeout(self) -> typing.Any:
		return self._instance.CommandTimeout
	@command_timeout.setter
	def command_timeout(self, value: typing.Any):
		self._instance.CommandTimeout = value
	@property
	def exit_status(self) -> int:
		return self._instance.ExitStatus
	@property
	def output_stream(self) -> typing.Any:
		return self._instance.OutputStream
	@property
	def extended_output_stream(self) -> typing.Any:
		return self._instance.ExtendedOutputStream
	@property
	def result(self) -> str:
		return self._instance.Result
	@property
	def error(self) -> str:
		return self._instance.Error

File: UnderAutomation.UniversalRobots.ObfuscatedSources/Ssh/Tools/SshCommand.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using System.IO;

namespace Ssh.Tools {
	/// <summary>
	/// Represents SSH command that can be executed.
	/// </summary>
	public class SshCommand : IDisposable {

		/// <summary>
		/// Gets the command text.
		/// </summary>
		public string CommandText { get; }

		/// <summary>
		/// Gets or sets the command timeout.
		/// </summary>
		/// <returns>The command timeout.</returns>
		public TimeSpan CommandTimeout { get; set; }

		/// <summary>
		/// Gets the command exit status.
		/// </summary>
		public int ExitStatus { get; }

		/// <summary>
		/// Gets the output stream.
		/// </summary>
		public Stream OutputStream { get; }

		/// <summary>
		/// Gets the extended output stream.
		/// </summary>
		public Stream ExtendedOutputStream { get; }

		/// <summary>
		/// Gets the command execution result.
		/// </summary>
		public string Result { get; }

		/// <summary>
		/// Gets the command execution error.
		/// </summary>
		public string Error { get; }

		/// <summary>
		/// Begins an asynchronous command execution.
		/// </summary>
		/// <returns>An <xref href="System.IAsyncResult" data-throw-if-not-resolved="false"></xref> that represents the asynchronous command execution, which could still be pending.</returns>
		public IAsyncResult BeginExecute()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Begins an asynchronous command execution.
		/// </summary>
		/// <param name="callback">An optional asynchronous callback, to be called when the command execution is complete.</param>
		/// <returns>An <xref href="System.IAsyncResult" data-throw-if-not-resolved="false"></xref> that represents the asynchronous command execution, which could still be pending.</returns>
		public IAsyncResult BeginExecute(AsyncCallback callback)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Begins an asynchronous command execution.
		/// </summary>
		/// <param name="callback">An optional asynchronous callback, to be called when the command execution is complete.</param>
		/// <param name="state">A user-provided object that distinguishes this particular asynchronous read request from other requests.</param>
		/// <returns>An <xref href="System.IAsyncResult" data-throw-if-not-resolved="false"></xref> that represents the asynchronous command execution, which could still be pending.</returns>
		public IAsyncResult BeginExecute(AsyncCallback callback, object state)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Begins an asynchronous command execution.
		/// </summary>
		/// <param name="commandText">The command text.</param>
		/// <param name="callback">An optional asynchronous callback, to be called when the command execution is complete.</param>
		/// <param name="state">A user-provided object that distinguishes this particular asynchronous read request from other requests.</param>
		/// <returns>An <xref href="System.IAsyncResult" data-throw-if-not-resolved="false"></xref> that represents the asynchronous command execution, which could still be pending.</returns>
		public IAsyncResult BeginExecute(string commandText, AsyncCallback callback, object state)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Waits for the pending asynchronous command execution to complete.
		/// </summary>
		/// <param name="asyncResult">The reference to the pending asynchronous request to finish.</param>
		/// <returns>Command execution result.</returns>
		public string EndExecute(IAsyncResult asyncResult)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Executes command specified by <xref href="UnderAutomation.UniversalRobots.Ssh.Tools.SshCommand.CommandText" data-throw-if-not-resolved="false"></xref> property.
		/// </summary>
		/// <returns>Command execution result</returns>
		public string Execute()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Cancels command execution in asynchronous scenarios.
		/// </summary>
		public void CancelAsync()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Executes the specified command text.
		/// </summary>
		/// <param name="commandText">The command text.</param>
		/// <returns>Command execution result</returns>
		public string Execute(string commandText)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
		/// </summary>
		public void Dispose()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Releases unmanaged and - optionally - managed resources
		/// </summary>
		/// <param name="disposing"><code>true</code> to release both managed and unmanaged resources; <code>false</code> to release only unmanaged resources.</param>
		protected virtual void Dispose(bool disposing)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}
	}
}
